File: server/app/api/sentiment.py
from flask import Blueprint, request, jsonify
from app.models.sentiment_model import analyze_sentiment

sentiment_blueprint = Blueprint('sentiment', __name__, url_prefix='/sentiment')

@sentiment_blueprint.route('/analyze', methods=['POST'])
def analyze():
    content = request.json
    text = content.get('text')
    sentiment_result = analyze_sentiment(text)
    return jsonify(sentiment=sentiment_result)

File: client/src/lib/sentiment.js
// lib/sentiment.js
function analyzeSentiment(text) {
  return fetch('http://localhost:5000/sentiment/analyze', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ text: text }),
  })
  .then(response => response.json())
  .then(data => data.sentiment);
}
